core/src/main/scala/filodb.core/query/QueryContext.scala

```diff
@@ -22,26 +22,50 @@ case class PromQlQueryParams(promQl: String, startSecs: Long, stepSecs: Long, en
 
 case object UnavailablePromQlQueryParams extends TsdbQueryParams
 
+case class IndividualQuota(
+        execPlanSamples: Int = 1000000,       // Limit on ExecPlan results in samples, default is 100K
+        execPlanResultBytes: Long = 18000000, // Limit on ExecPlan results in bytes, default is 18MB
+        groupByCardinality: Int = 100000,     // Limit on "group by" clause results, default is 100K
+        joinQueryCardinality: Int = 100000,   // Limit on binary join results, default is 100K
+        timeSeriesSamplesScanBytes: Long = 300000000)  // Limit on max data scanned per shard, default is 300 MB
+
+object IndividualQuota {
+  def apply(warnDefaults: Boolean): IndividualQuota = {
+    if (warnDefaults) {
+      IndividualQuota(
+        execPlanSamples = 50000,
+        execPlanResultBytes = 15000000,
+        groupByCardinality = 50000,
+        joinQueryCardinality = 50000,
+        timeSeriesSamplesScanBytes = 150000
+      )
+    } else {
+      IndividualQuota()
+    }
+  }
+}
```

Comment:
Are these separate "warning" thresholds needed? Is it sufficient to warn at X% of the limit?

If the separate thresholds are needed, a `Threshold(warning, limit)` class is another option to consider (this way we wouldn't need something like `apply(warnDefaults)`).

Reply:
I'm only copying whatever was already in the existing properties. This PR essentially groups limits in an object and allows to have two sets of quotas/limits: (1) enforced (2) warn and allows to pass these parameters from the QueryService but it DOES NOT change the existing behavior.

Reply:
The warning piece is fundamental for the PR. The reason we do not enforce the limits (we have very high limits) is because it's impossible for us to enforce them. There would be a number of users whose queries will stop to work and we simply cannot let it happen (imagine alerts or recording rules), instead, we need to:
A) establish quotas that we believe are resonable on WARN level
B) find out all the users/cases where the limits are breached
C) establish SPECIFIC HIGH limits for recording rules/alerts/ some API queries
D) enforce reasonable limits for EVERYBODY but the specific cases above
E) notify the users with specific high limits to let them know that their RR/alerts will be disabled in X days if they don't take care of it.

At this stage I think it's reasonable to have more flexibility in defining what the warn and enforce levels are. The ratio is hard to come up by looking at the crystal ball. Maybe it's 5x maybe 2. This we will figure out as part of an experiment around steps A/B/C.